src/lib.rs

```diff
@@ -144,6 +147,35 @@ where
         }
     }
 
+    #[cfg(feature = "std")]
+    pub fn get_firmware_version(&mut self) -> nb::Result<String, Error<E>> {
```

Comment:
I think it might be preferable to return `[u8; 4]` instead of `String`. It would avoid the dependency on `std`, and avoids the heap allocation.

Reply:
Makes more sense indeed, updated